src/stdlib/native/int.no

```diff
@@ -18,10 +19,90 @@ fn Uint8.new(val: Field) -> Uint8 {
     };
 }
 
+// u16
+struct Uint16 {
+    inner: Field,
+    bit_len: Field,
+}
+
+fn Uint16.new(val: Field) -> Uint16 {
+    let bit_len = 16;
+
+    // range check
+    let ignore_ = bits::to_bits(bit_len, val);
+
+    return Uint16 {
+        inner: val,
+        bit_len: bit_len
+    };
+}
+
+// u32
+struct Uint32 {
+    inner: Field,
+    bit_len: Field,
+}
+
+fn Uint32.new(val: Field) -> Uint32 {
+    let bit_len = 32;
+
+    // range check
+    let ignore_ = bits::to_bits(bit_len, val);
+
+    return Uint32 {
+        inner: val,
+        bit_len: bit_len
+    };
+}
+
+// u64
+struct Uint64 {
+    inner: Field,
+    bit_len: Field,
+}
+
+fn Uint64.new(val: Field) -> Uint64 {
+    let bit_len = 64;
+
+    // range check
+    let ignore_ = bits::to_bits(bit_len, val);
+
+    return Uint64 {
+        inner: val,
+        bit_len: bit_len
+    };
+}
+
+// implement comparator
+
 fn Uint8.less_than(self, rhs: Uint8) -> Bool {
     return comparator::less_than(self.bit_len, self.inner, rhs.inner);
 }
 
 fn Uint8.less_eq_than(self, rhs: Uint8) -> Bool {
     return comparator::less_eq_than(self.bit_len, self.inner, rhs.inner);
+}
+
+fn Uint16.less_than(self, rhs: Uint16) -> Bool {
+    return comparator::less_than(self.bit_len, self.inner, rhs.inner);
+}
+
+fn Uint16.less_eq_than(self, rhs: Uint16) -> Bool {
+    return comparator::less_eq_than(self.bit_len, self.inner, rhs.inner);
+}
+
+fn Uint32.less_than(self, rhs: Uint32) -> Bool {
+    return comparator::less_than(self.bit_len, self.inner, rhs.inner);
+}
+
+fn Uint32.less_eq_than(self, rhs: Uint32) -> Bool {
+    return comparator::less_eq_than(self.bit_len, self.inner, rhs.inner);
+}
+
+fn Uint64.less_than(self, rhs: Uint64) -> Bool {
+    return comparator::less_than(self.bit_len, self.inner, rhs.inner);
+}
+
+fn Uint64.less_eq_than(self, rhs: Uint64) -> Bool {
+    return comparator::less_eq_than(self.bit_len, self.inner, rhs.inner);
```

Comment:
we should also implement `add`, `sub`, etc.
this is when having a `bit_len` field would be interesting... because theoretically we could track how many bits max the result of an operation is, and only trigger a range check when we might be above 8 bits for u8. But in the absence of that we can simply perform a range check after every operation

Reply:
To take it further, maybe we can make it support builtin operations, such as + -, by mapping these operations to these functions. https://github.com/zksecurity/noname/issues/214